diesel_tests/tests/filter.rs

```diff
@@ -439,6 +439,101 @@ fn filter_subselect_with_boxed_query() {
     assert_eq!(expected, data);
 }
 
+#[test]
+fn filter_subselect_with_nullable_column() {
+    use schema_dsl::*;
+    table!{
+        heros {
+            id -> Integer,
+            name -> Text,
+            home_world -> Nullable<Integer>,
+        }
+    }
+    table!{
+        home_worlds {
+            id -> Integer,
+            name -> Text,
+        }
+    }
+
+    #[derive(Debug, Queryable, PartialEq)]
+    struct Hero {
+        id: i32,
+        name: String,
+        home_world: Option<i32>,
+    }
+    let connection = connection();
+
+    create_table(
+        "home_worlds",
+        (
+            integer("id").primary_key().auto_increment(),
+            string("name").not_null(),
+        ),
+    ).execute(&connection)
+        .unwrap();
+
+    create_table(
+        "heros",
+        (
+            integer("id").primary_key().auto_increment(),
+            string("name").not_null(),
+            integer("home_world"),
+        ),
+    ).execute(&connection)
+        .unwrap();
+
+    connection
+        .execute("INSERT INTO home_worlds (name) VALUES ('Tatooine');")
+        .unwrap();
+    connection
+        .execute(
```

Comment:
Let's avoid using private API in new tests. `sql_query` if we must, but this can just use the normal insert API